spec/helpers/oauth_controller_spec_helper.rb

```diff
@@ -0,0 +1,67 @@
+module OAuthControllerSpecHelper
+  include ActiveJob::TestHelper
+  def login_as_application_owner
+    @user = User.find_by_id(1) || create(:user)
+    @user.save!
+    login_as(@user)
+  end
+  
+  def setup_oauth
+    @server=OAuth::Server.new "http://localhost:3000"
+    @consumer=OAuth::Consumer.new('key','secret',{:site=>"http://localhost:3000"})
+
+    @client_application = ClientApplication.first
+    if @client_application == nil
+      @client_application = ClientApplication.new :user => @user, :key => @consumer.key, :secret => @consumer.secret
+      @client_application.name = "Client Application name"
+      @client_application.url = "http://localhost/"
+      @client_application.callback_url = "http://localhost:3000/callback"
+      @client_application.save!
+    end
+    @client_applications=[@client_application]
+    @current_client_application = @client_application
+    @current_client_applications = @client_applications
+    @request_token=RequestToken.create :user => @user, :client_application => @client_application, :callback_url=>@client_application.callback_url
+    
+    @access_token=AccessToken.create :user => @user, :client_application => @client_application
+  end
+  
+  def setup_oauth_for_user
+    login_as_application_owner
+    setup_oauth
+    @tokens=[@request_token]
+    @user.reload
+    allow(@user).to receive(:client_applications).and_return(@client_applications)
```

Comment:
(Mentioned by PM too) this stubbing is weird and probably the cause of the errors in RSpec – I'm betting that the model the application loads is a new User model that conceptually represents the same ID, so the stub won't go through. We should save these applications in the database where possible – you mentioned some errors with that, and I think it'd be good to get those straightened out